### PR TITLE
#302 Prevent new extras and parts for canceled subscriptions

### LINKED ISSUE #302: Active but canceled subscriptions should not get new extras or parts assigned
If a member has an active but canceled subscription with active but canceled extra subscriptions, adds a new future subscription (type) for next year and then wants to add future extra subscriptions, those (in some cases) get assigned to the (old) active but canceled subscription and not to the future subscription.

In general, it should not be possible to add extra subscriptions to canceled subscriptions. New extra subscriptions should get assigned to active or future subscriptions only. Or we generally let the member choose where to add.

### PR DESCRIPTION
**Pull request by nixnuex**
Close #302

### CODE REVIEW

File: juntagrico/views_subscription.py [hunk 1]
```diff
@@ -165,7 +165,8 @@ def size_change(request, subscription_id):
     change the size of a subscription
     """
     subscription = get_object_or_404(Subscription, id=subscription_id)
-    if request.method == 'POST':
+    parts_order_allowed = subscription.waiting or subscription.active
+    if request.method == 'POST' and parts_order_allowed:
```
Comment (motu55) [inline juntagrico/views_subscription.py:169]: throw an exception when someone orders a part in a cancelled subscription

File: juntagrico/views_subscription.py [hunk 2]
```diff
@@ -188,7 +190,8 @@ def extra_change(request, subscription_id):
     change an extra subscription
     '''
     subscription = get_object_or_404(Subscription, id=subscription_id)
-    if request.method == 'POST':
+    extra_order_allowed = subscription.waiting or subscription.active
+    if request.method == 'POST' and extra_order_allowed:
```
Comment (motu55) [inline juntagrico/views_subscription.py:194]: same here

**Review by motu55 — CHANGES_REQUESTED** [review]
all in all ok
also it seems that this pullrequest revertes this 
https://github.com/juntagrico/juntagrico/pull/292/files
the zusatzabo naming

**Comment by nixnuex** [comment]
True. I found more occurances of "Zusatzabo" than "Zusatz-Abo". Thats why. Will clean up the rest in a separate pull request.

**Review by motu55 — APPROVED** [review]

**Pull request merged**